include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -135,6 +135,16 @@ struct OrtEpApi {
   ORT_API_T(const char*, NodeComputeContext_NodeName, _In_ const OrtNodeComputeContext* context);
 };
 
+/**
+ * \brief The data layout type that is preferred by an EP.
+ * \since Version 1.23.
+ */
+typedef enum OrtEpDataLayout {
+  NCHW = 0,
+  NHWC,
+  NCHWC,
```

Comment:
I could be wrong, but I think `NCHWC` is not used by EPs (or maybe only used for L2 opts?). If so, should we remove this for now?